src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+		}
+
+		const cc = new ColorSwitchCCStartLevelChange(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent,
+			down: direction === "down",
+			startLevel: options.startLevel,
+		});
+
+		await this.driver.sendCommand(cc);
+	}
+
+	public async stopLevelChange(colorKey: ColorKey): Promise<void> {
+		this.assertSupportsCommand(
+			ColorSwitchCommand,
+			ColorSwitchCommand.StopLevelChange,
+		);
+
+		const colorComponent = getColorComponentFromKey(colorKey);
+		if (colorComponent == undefined) {
+			throw new Error(`Unknown color "${colorKey}".`);
+		}
+
+		const cc = new ColorSwitchCCStopLevelChange(this.driver, {
+			nodeId: this.endpoint.nodeId,
+			endpoint: this.endpoint.index,
+			colorComponent,
+		});
+
+		await this.driver.sendCommand(cc);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property, propertyKey },
+		value,
+	) => {
+		if (property !== "targetColor") {
+			throwUnsupportedProperty(this.ccId, property);
+		}
+		if (typeof value !== "number") {
+			throwWrongValueType(this.ccId, property, "number", typeof value);
+		}
+
+		if (!propertyKey) {
+			// No error throw util for this?
+			// Might want to treat keyless value as hex "#wwccrrggbb"
+			throw new Error("propertyKey required.");
+		}
+
+		if (typeof propertyKey != "string" || !isColorKey(propertyKey)) {
+			throwUnsupportedPropertyKey(this.ccId, property, propertyKey);
+		}
+
+		await this.set({ [propertyKey]: value });
+
+		// Refresh the current value
+		await this.get(propertyKey);
+	};
+}
+
+@commandClass(CommandClasses["Color Switch"])
+@implementedVersion(3)
+export class ColorSwitchCC extends CommandClass {
+	declare ccCommand: ColorSwitchCommand;
+
+	public async interview(complete: boolean = true): Promise<void> {
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const api = endpoint.commandClasses["Color Switch"];
+
+		log.controller.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: `${this.constructor.name}: doing a ${
+				complete ? "complete" : "partial"
+			} interview...`,
+			direction: "none",
+		});
+
+		// TODO: Only do this if complete?  Where would we get
+		//	the previously discovered colors?
+		log.controller.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: "querying Color Switch CC supported colors...",
+			direction: "outbound",
+		});
+
+		const colorsResponse = await api.getSupported();
+
+		log.controller.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: "querying Color Switch CC color states...",
+			direction: "outbound",
+		});
+
+		for (const key of ColorKeys) {
+			const supportsKey = supportedColorTableKey(key);
+			if (!colorsResponse[supportsKey]) {
+				continue;
+			}
+
+			await api.get(key);
+		}
+	}
+}
+
+@CCCommand(ColorSwitchCommand.SupportedReport)
+export class ColorSwitchCCSupportedReport extends ColorSwitchCC
+	implements Partial<SupportedColorTable> {
```

Comment:
Is this really Partial? In the API getSupported method you assert all properties to be non-null.

Reply:
For some reason I thought the data was parsed after the constructor call, so I tried to set it up to allow the data to be in an unparsed state.  I will remove the optionality.